### PR TITLE
fix configmap creation; add bad post create cfg change test cases

### PR DESCRIPTION
**Pull request by gabemontero**
@bparees ptal

### CODE REVIEW

File: pkg/stub/handler.go [hunk 1]
```diff
@@ -154,7 +158,7 @@ func (h *Handler) SpecValidation(srcfg *v1alpha1.SamplesResource) error {
 		return nil
 	}
 	cm, err := h.configmapclientwrapper.Get(h.namespace, v1alpha1.SamplesResourceName)
-	if cm == nil || err != nil {
+	if err != nil && !kerrors.IsNotFound(err) {
```
Comment (bparees) [inline pkg/stub/handler.go]: if it is an IsNotFound err, i think you need to make sure cm gets set to a non-nil value.  I wouldn't count on get always returning an empty one (i find it weird that it ever does)

Comment (bparees) [reply pkg/stub/handler.go]: also i'm not sure what you expect to do if it is an IsNotFoundErr?

File: pkg/stub/handler.go [hunk 2]
```diff
@@ -114,6 +114,10 @@ func (h *Handler) StatusUpdate(condition v1alpha1.SamplesResourceConditionType,
 			// just return error to sdk for retry
 			return err
 		}
+		if kerrors.IsNotFound(err) {
+			// 4.0 testing showed that we were getting emtpy ConfigMaps
```
Comment (bparees) [inline pkg/stub/handler.go]: typo in empty

File: pkg/stub/handler.go (hunk 1 above)
Comment (gabemontero) [reply pkg/stub/handler.go]: Yeah need to error out with a ConditionUnknown status for the ConfigurationValid condition ... presumably somebody deleted the configmap underneath us, or some such pre-existing problem

**Comment by bparees** [comment]
lgtm

**Pull request merged**